launcher/modplatform/modrinth/ModrinthPackExportTask.cpp

```diff
@@ -290,30 +290,40 @@ QByteArray ModrinthPackExportTask::generateIndex()
         if (forge != nullptr)
             dependencies["forge"] = forge->m_version;
 
-        obj["dependencies"] = dependencies;
+        out["dependencies"] = dependencies;
     }
 
-    QJsonArray files;
-    QMapIterator<QString, ResolvedFile> iterator(resolvedFiles);
-    while (iterator.hasNext()) {
-        iterator.next();
+    QJsonArray filesOut;
+    for (auto iterator = resolvedFiles.constBegin(); iterator != resolvedFiles.constEnd(); iterator++) {
+        QJsonObject fileOut;
 
+        QString path = iterator.key();
         const ResolvedFile& value = iterator.value();
 
-        QJsonObject file;
-        file["path"] = iterator.key();
-        file["downloads"] = QJsonArray({ iterator.value().url });
+        // detect disabled mod
+        QString disabledSuffix = ".disabled";
+        if (path.endsWith(disabledSuffix)) {
+            // rename it
+            path = path.left(path.length() - disabledSuffix.length());
```

Comment:
this might be a more simple way to do it
```suggestion
        QString disabledSuffix = "disabled";
        QFileInfo pathFile(path);
        if (pathFile.suffix() == disabledSuffix) {
            // rename it
            path = pathFile.completeBaseName();
```

Reply:
^^

Reply:
I actually had a good look at the implementation of QFileInfo's constructor
this shouldn't affect performance...  :laughing:

Reply:
Won't work

https://doc.qt.io/qt-6/qfileinfo.html#completeBaseName

Reply:
> I actually had a good look at the implementation of QFileInfo's constructor this shouldn't affect performance... 😆

it wasn't really intended to. it just makes everything a lot more readable

> Won't work

it will - unless i missed something here and the path isn't meant to be a QString. an example is:
```cpp
#include <QString>
#include <QList>
#include <QFileInfo>
#include <iostream>

int main() {
        QList<QString> paths = {"mod1.jar", "mod2.jar.disabled"};
        std::cout << "original files are:\n" << std::endl;

        for (const QString &path : paths) {
                std::cout << path.toStdString() << "\n" << std::endl;
        }

        std::cout << "new files are:\n" << std::endl;

        QString disabledSuffix = "disabled";

        for (QString &path : paths) {
                QFileInfo pathFile(path);
                if (pathFile.suffix() == disabledSuffix) {
                        path = pathFile.completeBaseName();
                }
                std::cout << path.toStdString() << "\n" << std::endl;
        }
}
```

which outputs

```
original files are:

mod1.jar

mod2.jar.disabled

new files are:

mod1.jar

mod2.jar
```

Reply:
I meant I was worried about performance for no reason

Anyway, it won't work as it'll remove the path

Reply:
ohhh you want the full path! massive brain fart

```cpp
path = pathFile.dir().absolutePath() + "/" + pathFile.completeBaseName();
```
would work then

Reply:
at this point I don't think it's cleaner

Reply:
what do you think?

Reply:
i still think using QFileInfo methods for file paths over substring manipulation makes a little more sense and could help out if this ever gets expanded. splitting the variables might help here with something like

```cpp
QString pathDir = pathFile.dir().absolutePath();
path = pathDir + "/" + path.completeBaseName();
```

`canonicalPath()` could also probably be used instead of the QDir methods, since the directory is pretty much guaranteed to exist anyways - and it would get rid of the slightly silly `.dir().absolutePath()`

Reply:
Hmm... while it's not cleaner it's more logical... maybe

Reply:
Still won't work. We don't want the absolute path.

Reply:
`pathInfo.dir().filePath(pathInfo.completeBaseName())`

seems to work

Reply:
sou could also use `pathInfo.dir().absoluteFilePath(pathInfo.completeBaseName())`

Reply:
Nope